### PR TITLE
Volume attachments in sequence to an instance

### LINKED ISSUE #2390: bug: ibm_is_volume are not attached in sequence to ibm_is_instance
### Terraform Version
```
[:~] $ terraform -v
Terraform v0.14.8
```

### Affected Resource(s)

- ibm_is_instance
- ibm_is_volume

### Terraform Configuration Files
```hcl

resource "ibm_is_volume" "virtual_server_block_volume_data_1" {
  name     = "${var.virtual_server_hostname}-volume-data-1"
  resource_group = var.target_resource_group_id
  zone     = var.target_availability_zone
  profile  = "custom"
  iops     = 1000
  capacity = 100
}

resource "ibm_is_volume" "virtual_server_block_volume_data_2" {
  name     = "${var.virtual_server_hostname}-volume-data-2"
  resource_group = var.target_resource_group_id
  zone     = var.target_availability_zone
  profile  = "custom"
  iops     = 1000
  capacity = 200
}

resource "ibm_is_volume" "virtual_server_block_volume_data_3" {
  name     = "${var.virtual_server_hostname}-volume-data-3"
  resource_group = var.target_resource_group_id
  zone     = var.target_availability_zone
  profile  = "custom"
  iops     = 1000
  capacity = 300
}

resource "ibm_is_volume" "virtual_server_block_volume_data_4" {
  name     = "${var.virtual_server_hostname}-volume-data-4"
  resource_group = var.target_resource_group_id
  zone     = var.target_availability_zone
  profile  = "custom"
  iops     = 1000
  capacity = 400
}

resource "ibm_is_volume" "virtual_server_block_volume_data_5" {
  name     = "${var.virtual_server_hostname}-volume-data-5"
  resource_group = var.target_resource_group_id
  zone     = var.target_availability_zone
  profile  = "custom"
  iops     = 1000
  capacity = 500
}

resource "ibm_is_volume" "virtual_server_block_volume_data_6" {
  name     = "${var.virtual_server_hostname}-volume-data-6"
  resource_group = var.target_resource_group_id
  zone     = var.target_availability_zone
  profile  = "custom"
  iops     = 1000
  capacity = 600
}

resource "ibm_is_volume" "virtual_server_block_volume_data_7" {
  name     = "${var.virtual_server_hostname}-volume-data-7"
  resource_group = var.target_resource_group_id
  zone     = var.target_availability_zone
  profile  = "custom"
  iops     = 1000
  capacity = 700
}

resource "ibm_is_volume" "virtual_server_block_volume_data_8" {
  name     = "${var.virtual_server_hostname}-volume-data-8"
  resource_group = var.target_resource_group_id
  zone     = var.target_availability_zone
  profile  = "custom"
  iops     = 1000
  capacity = 800
}


resource "ibm_is_instance" "virtual_server" {

  depends_on = [
    ibm_is_volume.virtual_server_block_volume_data_1,
    ibm_is_volume.virtual_server_block_volume_data_2,
    ibm_is_volume.virtual_server_block_volume_data_3,
    ibm_is_volume.virtual_server_block_volume_data_4,
    ibm_is_volume.virtual_server_block_volume_data_5,
    ibm_is_volume.virtual_server_block_volume_data_6,
    ibm_is_volume.virtual_server_block_volume_data_7,
    ibm_is_volume.virtual_server_block_volume_data_8
    ]

  resource_group = var.target_resource_group_id
  name    = var.virtual_server_hostname
  profile = var.virtual_server_profile
  vpc  = var.target_vpc_id
  zone = var.target_availability_zone
  keys = [ibm_is_ssh_key.virtual_server_public_ssh.id]

  primary_network_interface {
    name = "${var.virtual_server_hostname}-0"
    subnet = data.ibm_is_subnet.vpc_subnet.id
  }

  boot_volume {
    name = "${var.virtual_server_hostname}-volume-boot-0"
  }

  volumes = [
    ibm_is_volume.virtual_server_block_volume_data_1.id,
    ibm_is_volume.virtual_server_block_volume_data_2.id,
    ibm_is_volume.virtual_server_block_volume_data_3.id,
    ibm_is_volume.virtual_server_block_volume_data_4.id,
    ibm_is_volume.virtual_server_block_volume_data_5.id,
    ibm_is_volume.virtual_server_block_volume_data_6.id,
    ibm_is_volume.virtual_server_block_volume_data_7.id,
    ibm_is_volume.virtual_server_block_volume_data_8.id
    ]

  //User can configure timeouts
  timeouts {
    create = "90m"
    delete = "30m"
  }

}
```

### Expected Behavior

Volumes created in sequence:
```shell
[root@virtualserver ~]# lsblk
NAME   MAJ:MIN RM  SIZE RO TYPE MOUNTPOINT
vda    252:0    0  100G  0 disk 
├─vda1 252:1    0    1G  0 part /boot
└─vda2 252:2    0   99G  0 part /
vdb    252:16   0  376K  0 disk 
vdc    252:32   0   44K  0 disk [SWAP]
vdd    252:48   0  100G  0 disk 
vde    252:64   0  200G  0 disk 
vdf    252:80   0  300G  0 disk 
vdg    252:96   0  400G  0 disk 
vdh    252:112  0  500G  0 disk 
vdi    252:128  0  600G  0 disk 
vdj    252:144  0  700G  0 disk 
vdk    252:160  0  800G  0 disk
```

### Actual Behavior

Volumes created out of sequence:
```shell
[root@virtualserver ~]# lsblk
NAME   MAJ:MIN RM  SIZE RO TYPE MOUNTPOINT
vda    252:0    0  100G  0 disk 
├─vda1 252:1    0    1G  0 part /boot
└─vda2 252:2    0   99G  0 part /
vdb    252:16   0  376K  0 disk 
vdc    252:32   0   44K  0 disk [SWAP]
vdd    252:48   0  500G  0 disk 
vde    252:64   0  700G  0 disk 
vdf    252:80   0  200G  0 disk 
vdg    252:96   0  800G  0 disk 
vdh    252:112  0  100G  0 disk 
vdi    252:128  0  600G  0 disk 
vdj    252:144  0  300G  0 disk 
vdk    252:160  0  400G  0 disk
```

### Steps to Reproduce

1. `terraform apply`

### PR DESCRIPTION
**Pull request by SunithaGudisagarIBM**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #2390 

Output from acceptance testing:

Screenshot 2023-03-14 at 1 15 18 PM
Screenshot 2023-03-14 at 2 46 18 PM
Screenshot 2023-03-14 at 3 24 51 PM
Screenshot 2023-03-14 at 3 35 28 PM
Screenshot 2023-03-14 at 10 20 06 AM
Screenshot 2023-03-14 at 12 28 04 PM

### CODE REVIEW

**Comment by kavya498** [comment]
@SunithaGudisagarIBM 
Can you please attach all test results of is_instance resource?

File: ibm/service/vpc/resource_ibm_is_instance.go [hunk 1]
```diff
@@ -4253,3 +4259,19 @@ func GetInstanceMetadataServiceOptions(d *schema.ResourceData) (metadataService
 	}
 	return nil
 }
+
+// This function takes two lists and returns the difference between the two lists
+// listdifference([1,2] [2,3]) = [1]
```
Comment (hkantare) [inline ibm/service/vpc/resource_ibm_is_instance.go:4264]: We already have this function https://github.com/IBM-Cloud/terraform-provider-ibm/blob/f7043dd036eefba25a2d8ca27dae0625910a0347/ibm/service/classicinfrastructure/resource_ibm_multi_vlan_firewall.go#L415
here also can we move to structure.go and use for both

Comment (hkantare) [reply ibm/service/vpc/resource_ibm_is_instance.go:4264]: yes just move to common place and update that also

Comment (SunithaGudisagarIBM1) [reply ibm/service/vpc/resource_ibm_is_instance.go:4613]: Changes incorporated

Comment (SunithaGudisagarIBM1) [reply ibm/service/vpc/resource_ibm_is_instance.go:4613]: Screenshot 2023-04-20 at 7 00 53 AM

**Comment by SunithaGudisagarIBM1** [comment]
Screenshot 2023-04-20 at 7 00 53 AM

**Comment by SunithaGudisagarIBM1** [comment]
Screenshot 2023-04-20 at 7 07 56 AM

**Pull request merged**